## NumPy Practice Exercise 1

Below, you are provided with 2 lists that contain the data of an ecommerce website.

* The first list contains the data for the number of items sold for a particular product
* The second list contains the price of one unit of the product sold

Each element in the list `number` shows the number of units sold for the product whose price is given at the same index in the list `price`.


As a part of this exercise, solve the questions that are provided below.

In [2]:
number = [8, 9, 9, 1, 6, 9, 5, 7, 3, 9, 7, 3, 4, 8, 3, 5, 8, 4, 8, 7, 5, 7, 3, 6, 1, 2, 7, 4, 7, 7, 8, 4, 3, 4, 2, 2, 2, 7, 3, 5, 6, 1, 1, 3, 2, 1, 1, 7, 7, 1, 4, 4, 5, 6, 1, 2, 7, 4, 5, 8, 1, 4, 8, 6, 2, 4, 3, 7, 3, 6, 2, 3, 3, 3, 2, 4, 6, 8, 9, 3, 9, 3, 1, 8, 6, 6, 3, 3, 9, 4, 6, 4, 9, 6, 7, 1, 2, 8, 7, 8, 1, 4]
price = [195, 225, 150, 150, 90, 60, 75, 255, 270, 225, 135, 195, 30, 15, 210, 105, 15, 30, 180, 60, 165, 60, 45, 225, 180, 90, 30, 210, 150, 15, 270, 60, 210, 180, 60, 225, 150, 150, 120, 195, 75, 240, 60, 45, 30, 180, 240, 285, 135, 165, 180, 240, 60, 105, 165, 240, 120, 45, 120, 165, 285, 225, 90, 105, 225, 45, 45, 45, 75, 180, 90, 240, 30, 30, 60, 135, 180, 15, 255, 180, 270, 135, 105, 135, 210, 180, 135, 195, 225, 75, 225, 15, 240, 60, 15, 180, 255, 90, 15, 150, 230, 150]

#### 1. How many different products are sold by the company in total?

1.  99
2. 100
3. 101
4. 102


In [3]:
# Type your code here

In [7]:
import numpy as np

In [9]:
import pandas as pd

# Example DataFrame with a categorical column
number = [8, 9, 9, 1, 6, 9, 5, 7, 3, 9, 7, 3, 4, 8, 3, 5, 8, 4, 8, 7, 5, 7, 3, 6, 1, 2, 7, 4, 7, 7, 8, 4, 3, 4, 2, 2, 2, 7, 3, 5, 6, 1, 1, 3, 2, 1, 1, 7, 7, 1, 4, 4, 5, 6, 1, 2, 7, 4, 5, 8, 1, 4, 8, 6, 2, 4, 3, 7, 3, 6, 2, 3, 3, 3, 2, 4, 6, 8, 9, 3, 9, 3, 1, 8, 6, 6, 3, 3, 9, 4, 6, 4, 9, 6, 7, 1, 2, 8, 7, 8, 1, 4]
price = [195, 225, 150, 150, 90, 60, 75, 255, 270, 225, 135, 195, 30, 15, 210, 105, 15, 30, 180, 60, 165, 60, 45, 225, 180, 90, 30, 210, 150, 15, 270, 60, 210, 180, 60, 225, 150, 150, 120, 195, 75, 240, 60, 45, 30, 180, 240, 285, 135, 165, 180, 240, 60, 105, 165, 240, 120, 45, 120, 165, 285, 225, 90, 105, 225, 45, 45, 45, 75, 180, 90, 240, 30, 30, 60, 135, 180, 15, 255, 180, 270, 135, 105, 135, 210, 180, 135, 195, 225, 75, 225, 15, 240, 60, 15, 180, 255, 90, 15, 150, 230, 150]
df = pd.DataFrame({'number': number, 'price': price})

# Describe the categorical column
print(df['number'].describe())
print(df['price'].describe())

count    102.000000
mean       4.803922
std        2.537010
min        1.000000
25%        3.000000
50%        4.000000
75%        7.000000
max        9.000000
Name: number, dtype: float64
count    102.000000
mean     139.019608
std       79.509758
min       15.000000
25%       60.000000
50%      150.000000
75%      210.000000
max      285.000000
Name: price, dtype: float64


#### 2. How many items were sold in total?

1. 460
2. 490
3. 500
4. 520

In [18]:
# Type your code here
print(df['number'].sum())
print(df['price'].sum())

490
14180


#### 3. What is the average price of the products sold by the ecommerce company?

1. 139
2. 151
3. 142
4. 128

In [14]:
# Type your code here
print(df['price'].mean())

139.01960784313727


#### 4. What is the price of the costliest item sold?

1. 225
2. 310
3. 280
4. 285

In [15]:
# Type your code here
print(df['price'].max())

285


#### 5. What is the total revenue of the company? [Revenue = Price\*Quantity]

1. 67100
2. 53900
3. 45300
4. 71200

In [19]:
# Type your code here
print((df['number']*df['price']).sum())

67100


#### 6. Demand for the 20th product in the list is more than the 50th product. [True/False]
1. True
2. False
3. Can't be calculated

In [21]:
# Type your code here
print(df['number'][20]>df['number'][50])
print(df['number'][19]>df['number'][49])

True
True


#### 7. How many products fall under the category of expensive goods?
An expensive good is that good whose price is more than the average price of the products sold by the company.

1. 48
2. 50
3. 52
4. 54

In [22]:
# Type your code here
print( df[df['price'].mean() < df['price']]['price'].count())

52
